### ISSUE TITLE
Consider showing a decoded version of the Certificate in Appendix A

### ISSUE THREAD

**Issue by jvanasco** [aarongable/draft-acme-ari#88]
A particular frustrating point I had in working on ARI is the spec's decision to utilize specific encodings in the logic example:

```

For example, to request renewal information for the end-entity certificate given in Appendix A, the client would make the request as follows:

1. The `keyIdentifier` field of the certificate's AKI extension has the hexadecimal bytes `69:88:5B:6B:87:46:40:41:E1:B3:7B:84:7B:A0:AE:2C:DE:01:C8:D4` as its ASN.1 Octet String value. The base64url encoding of those bytes is `aYhba4dGQEHhs3uEe6CuLN4ByNQ=`.
2. The certificate's Serial Number field has the hexadecimal bytes `00:87:65:43:21` as its DER encoding (note the leading zero byte to ensure the serial number remains positive despite the leading 1 bit in `0x87`). The base64url encoding of those bytes is `AIdlQyE=`.
3. Stripping the trailing padding characters and concatenating with the separator, the unique identifier is therefore `aYhba4dGQEHhs3uEe6CuLN4ByNQ.AIdlQyE`, and the client makes the request (split onto multiple lines for readability):
```

The problem is that when using certain libraries to handle the certificates, you are likely to get different data as the entire certificate is decoded into human text components - e.g. I get `2271560481` as the serial.  This creates added complexity when constructing a library and tests, as the example may need to be encoded/decoded further.

It would be useful if Appendix A contained a brief table outlining the decoded certificate so people wouldn't have to remember they may need to decode the cert to double check the data, and can reference all the possible return values within the RFC.  Alternately, noting the decoded value of the serial (`2271560481`) in the example would be incredibly useful as well. 

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 2271560481 (0x87654321)
        Signature Algorithm: ecdsa-with-SHA256
        Issuer: CN = Example CA
        Validity
            Not Before: Jan  1 00:00:00 1 GMT
            Not After : Jan  1 00:00:00 1 GMT
        Subject: CN = example.com
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:78:16:6e:ed:c6:e9:01:83:57:64:b6:db:87:ca:
                    cd:23:3b:8e:a2:a3:8e:b6:6c:40:d6:fe:dc:46:6f:
                    ff:03:0c:8c:c1:6c:72:1f:3e:33:7f:09:81:85:c4:
                    ab:7f:8e:cd:50:01:5f:ab:32:d0:d8:f3:d0:c5:d4:
                    d7:44:46:04:9c
                ASN1 OID: prime256v1
                NIST CURVE: P-256
        X509v3 extensions:
            X509v3 Authority Key Identifier: 
                69:88:5B:6B:87:46:40:41:E1:B3:7B:84:7B:A0:AE:2C:DE:01:C8:D4
    Signature Algorithm: ecdsa-with-SHA256
    Signature Value:
        30:44:02:20:7b:4f:7e:4b:94:d9:02:5c:39:b6:0b:62:56:fb:
        84:45:5e:9c:4f:89:9f:ba:d5:c8:97:04:db:f8:56:0d:ff:ca:
        02:20:29:43:b2:a0:64:96:10:7d:28:07:a5:16:26:2d:f7:ee:
        85:0e:3d:f3:e0:a1:95:7a:56:0c:e4:c7:6c:3b:7a:4a
```

**Comment by aarongable** [comment]
I'd like to understand this request better -- I consciously chose _not_ to include a textual representation of the certificate _because_ the textual representations of the fields we care about don't line up with the underlying byte representations that are used by the URL construction.

I guess I don't understand the circumstances in which knowing that the cert's serial is "2271560481" is useful to a client developer?

**Comment by jvanasco** [comment]
Use case:

I have a certificate and am following the RFC to construct the ARI url.  I parse the Certificate with the tools/libraries already used by my client, and try to build a test case around the RFC example - but nothing works. After doing some debugging, downloading more tools, and looking through the various API docs and code, I realize that I am getting a textual representation of `2271560481` as the output, because just about everything fully decodes the field (possibly everything).  Had I realized that sooner, I would have saved hours, as I would have immediately seen and understood the encoding issue.

**Comment by aarongable** [comment]
Sure, that flow makes sense... but I don't see how knowing _ahead of time_ that the textual representation of the serial is "2271560481" would have helped. The issue isn't not knowing what integer the serial decodes to, the issue is not knowing that the software is decoding the serial bytes to an integer.

**Comment by jvanasco** [comment]
> the issue is not knowing that the software is decoding the serial bytes to an integer

Yes.  Knowing the underlying libraries *might* decode the serial bytes to an integer would have been 95% useful here, knowing the exact integer just makes things more obvious.

When it comes to the AKID, the underlying libraries in Python also all have some weird behaviors; they seem to fully decode the serial but still leave the tag on - so you have to discard the first 4 bytes of the value.  That drove me batty for a bit.  `orangepizza` had figured that out in their [Certbot PR](https://github.com/certbot/certbot/pull/9945/files#diff-035393341b36c4adff76feb0627156920de618857c156c42335ef9b0437de6fe)

**Comment by aarongable** [comment]
> When it comes to the AKID, the underlying libraries in Python also all have some weird behaviors; they seem to fully decode the serial but still leave the tag on - so you have to discard the first 4 bytes of the value.

That's not a weird behavior, that's the library doing the right thing. It's returning [the whole authorityKeyIdentifier extension](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.1). ARI wants the keyIdentifier _field of_ that extension. The proper thing to do would be to parse those bytes as ASN.1 and access the field with tag 0. Simply discarding the first four bytes is likely to work most of the time in practice, but is by no means guaranteed.
